src/common/procedure/src/local.rs

```diff
@@ -492,6 +487,13 @@ impl LocalManager {
 
         let watcher = meta.state_receiver.clone();
 
+        ensure!(
+            self.manager_ctx.num_running_procedures() < self.config.max_running_procedures,
+            TooManyRunningProceduresSnafu {
+                max_running_procedures: self.config.max_running_procedures,
+            }
+        );
```

Comment:
When will we have so many running procedures?

Reply:
When the metric engine is enabled with remote WAL, if remote WAL becomes unavailable, the upper layer continues to return a "storage unavailable" error. In procedures, such errors trigger retries. Since automatic table creation keeps failing, write operations continuously trigger the auto-create table procedure. Eventually, this leads to metasrv running out of memory